README
$Id: README,v 1.1 2010/05/31 03:53:02 ak Exp $
--------------------------------------------------------------------------------
c2p is a converter which convert from character(string) to phonetic code.
--------------------------------------------------------------------------------

* HOW TO BUILD
    * Default language is English
    # make

* INSTALL(->/usr/local/bin)
    # make install

* UNINSTALL
    # make uninstall

* USAGE
    > c2p -h
      Shows help message of this commnad

    > c2p Secret
      Sierra(LARGE),Echo,Charlie,Romeo,Echo,Tango

    > c2p -dw Password
      Papa(LARGE) Alfa Sierra Sierra Whiskey Oscar Romeo Delta

    > echo 'String' | c2p
      Sierra(LARGE),Tango,Romeo,India,November,Golf



CHANGES
-----------------------------------------------------------------------
1.1.8   2013-08-31
    * Indent style has been updated: 8 width tab => 4 space characters.

1.1.7   2012-08-14
    * Add ``chmod a+x'' in Makefile.

1.1.6   2010-06-02
    * Remove 2 options: -e and -j completely.
    * Update help message.
    * Add README.JA

1.1.5   2010-05-31
    * Initialize the pointer: **selected_language = NULL;
    * Add language file: jp.h and en.h
    * Remove unused variable.
    * Remove 2 options: -e and -j. Use -Lx option.
    * Default language is English.

1.1.4   2009-08-09
    * Add the new option '-L', select displaying language(e or j)
    * Japanese characters are written by UTF-8.

1.1.3   2009-04-29
    * New macro 'DEFAULT_LANGUAGE'
    * New compile time macro 'DAIJI'

1.1.2   2009-02-11
    * Correct spell('Hyphen') in English mode.
    * Add the new option '-dh': Use '-' as a delimiter.

1.1.1   2008-10-18
    * fprintf() typecasts to the void.

1.1.0   2008-04-01
    * Commit in Hokkaido !
    * Remove the option -x(debug mode).

1.0.15  2008-02-17
    * Add description of '-C' option in the help screen.
    * The number 32(dec) is replaced by the character ' '(Space).

1.0.14  2008-02-11
    * Add the new option '-C': shows control character.
    * Fix for the case that includes multibyte character.

1.0.13  2008-01-27
    * Fix usage of the following functions, strlen(), strlcpy() .

1.0.12  2008-01-26
    * Use the following string functions: isprint(), iscntrl(), and
      tolower().

1.0.11  2007-12-30
    * Check the length of the string while read from stdin.

1.0.10  2007-12-29
    * c2p recieves string from STDIN.

1.0.9   2006-09-15
    * Add the new compile macro 'KANJI'.  Some symbols and numbers
      are output as KANJI.
    * Add another pronunciations in comment.

1.0.8   2006-06-02
    * Uses EXIT_SUCCESS or EXIT_FAILURE (stdlib.h) at exit status.

1.0.7   2004-05-05
    * Support white spaces in the argument.

1.0.6   2004-04-29
    * Fix: Skip if control character was appeared.
    * Fix: Skip if multibyte character was appeared.
    * Fix: Exit if invalid option was specified.

1.0.5   2003-01-07
    * Add the New option '-s' to write a white space after a delim-
      iter.
    * Add pre-processor for the system that has no 'strlcpy'

1.0.4   2002-12-30
    * Add the new delimiter ' ': White space.

1.0.3   2002-12-23
    * Fixed declaration of some variables to compile with gcc 2.96.

1.0.2   1997-10-01
    * Add the new option -d: Delimiter.

1.0.1   1997-08-31
    * Add the new function help() and the new option '-h'.
    * Add the new option -v: to print the version number.

1.0.0   1997-04-29
    * Initial version: c2p is a converter which  convert from char-
      acter(string) to phonetic code.
-----------------------------------------------------------------------


jp.h
/* jp.h for Japanese */
const char *PHONETIC_JAPANESE[] = {
    "(大)",                 /* IS UPPER CASE */
    "[ヘッダ開始]",         /* 01   SOA */
    "[テキスト開始]",       /* 02   STX */
    "[テキスト終了]",       /* 03   ETX */
    "[転送終了]",           /* 04   EOT */
    "[問い合わせ]",         /* 05   ENQ */
    "[肯定応答]",           /* 06   ACK */
    "[ベル]",               /* 07   BEL */
    "[1文字後退]",          /* 08   BS */
    "[水平タブ]",           /* 09   HT */
    "[改行]",               /* 10   LF */
    "[垂直タブ]",           /* 11   VT */
    "[改頁]",               /* 12   FF */
    "[復帰]",               /* 13   CR */
    "[シフトアウト]",       /* 14   SO */
    "[シフトイン]",         /* 15   SI */
    "[データリンク拡張]",   /* 16   DLE */
    "[装置制御1/XON]",      /* 17   DC1 */
    "[装置制御2]",          /* 18   DC2 */
    "[装置制御3/XOFF]",     /* 19   DC3 */
    "[装置制御4]",          /* 20   DC4 */
    "[否定応答]",           /* 21   NAK */
    "[同期信号]",           /* 22   SYN */
    "[ブロック転送終了]",   /* 23   ETB */
    "[キャンセル]",         /* 24   CAN */
    "[媒体終端]",           /* 25   EM */
    "[置換]",               /* 26   SUB */
    "[エスケープ]",         /* 27   ESC */
    "[ファイルセパレータ]", /* 28   FS */
    "[グループセパレータ]", /* 29   GS */
    "[レコードセパレータ]", /* 30   RS */
    "[ユニットセパレータ]", /* 31   US */
    "空白",             /* 32   SPACE */
    "びっくり",         /* 33   ! [感嘆符,雨垂れ,エクスクラメーションマーク] */
    "二重引用府",       /* 34   " [引用符] */
#ifdef KANJI
    "井",               /* 35   # */
    "弗",               /* 36   $ */
#else
    "シャープ",         /* 35   # [井] */
    "ドル",             /* 36   $ [弗] */
#endif
    "パーセント",       /* 37   % [百分率] */
    "アンド",           /* 38   & [アンパサンド] */
    "引用府",           /* 39   ' [アポストロフィー] */
    "左丸括弧",         /* 40   ( [パーレン] */
    "右丸括弧",         /* 41   ) */
#ifdef KANJI
    "米",               /* 42   * [星,水] */
    "足す",             /* 43   + [プラス] */
#else
    "アステリスク",     /* 42   * [星号] */
    "プラス",           /* 43   + [プラス] */
#endif
    "カンマ",           /* 44   , [コンマ] */
#ifdef KANJI
    "引く",             /* 45   - [マイナス,ハイフン] */
    "点",               /* 46   . [ドット,ピリオド] */
    "斜線",             /* 47   / [スラッシュ,スラント,ソリダス] */
#else
    "マイナス",         /* 45   - */
    "ドット",           /* 46   . */
    "スラッシュ",       /* 47   / */
#endif
    "零",               /* 48   0 */
#ifdef DAIJI
    "壹",               /* 49   1 */
    "貮",               /* 50   2 */
    "參",               /* 51   3 */
#elif KANJI
    "壱",               /* 49   1 */
    "弐",               /* 50   2 */
    "参",               /* 51   3 */
#else
    "一",               /* 49   1 */
    "二",               /* 50   2 */
    "三",               /* 51   3 */
#endif
#ifdef DAIJI
    "肆",               /* 52   4 */
    "伍",               /* 53   5 */
    "陸",               /* 54   6 */
    "柒",               /* 55   7 */
    "捌",               /* 56   8 */
    "玖",               /* 57   9 */
#else
    "四",               /* 52   4 */
    "五",               /* 53   5 */
    "六",               /* 54   6 */
    "七",               /* 55   7 */
    "八",               /* 56   8 */
    "九",               /* 57   9 */
#endif
    "コロン",           /* 58   : */
    "セミコロン",       /* 59   ; */
    "左山括弧",         /* 60   < [小なり] */
    "等号",             /* 61   = [イコール] */
    "右山括弧",         /* 62   > [大なり] */
    "はてな",           /* 63   ? [疑問符] */
#ifdef KANJI
    "単価",             /* 64   @ [アットマーク] */
#else
    "アットマーク",     /* 64   @ [アットマーク] */
#endif
    "左角括弧",         /* 91   [ [ブラケット] */
    "円",               /* 92   Yen */
    "右角括弧",         /* 93   ] */
    "山",               /* 94   ^ [サーカムフレックス] */
    "下線",             /* 95   _ [アンダースコア] */
    "バッククォート",   /* 96   ` [アクサングラーヴ,グレイヴ・アクセント] */
    "エイ",             /* 97   a */
    "ビー",             /* 98   b */
    "スィー",           /* 99   c */
    "ディー",           /* 100  d */
    "イー",             /* 101  e */
    "エフ",             /* 102  f */
    "ジー",             /* 103  g */
    "エイチ",           /* 104  h */
    "アイ",             /* 105  i */
    "ジェイ",           /* 106  j */
    "ケー",             /* 107  k */
    "エル",             /* 108  l */
    "エム",             /* 109  m */
    "エヌ",             /* 110  n */
    "オー",             /* 111  o */
    "ピー",             /* 112  p */
    "キュー",           /* 113  q */
    "アール",           /* 114  r [尺] */
    "エス",             /* 115  s */
    "ティー",           /* 116  t [丁] */
    "ユー",             /* 117  u */
    "ブイ",             /* 118  v */
    "ダブリュ",         /* 119  w */
    "エックス",         /* 120  x */
    "ワイ",             /* 121  y */
    "ゼット",           /* 122  z [乙] */
    "左中括弧",         /* 123  { [中括弧,ブレイス] */
    "パイプ",           /* 124  | */
    "右中括弧",         /* 125  } */
#ifdef KANJI
    "波",               /* 126  ~ [波線] */
#else
    "チルダ",           /* 126  ~ [波線] */
    "[削除]",           /* 127  DEL */
#endif
}; /* End of PHONETIC_JAPANESE */



c2p.c
/* $Id: c2p.c,v 1.29 2010/05/31 02:20:12 ak Exp $ */
/*                                                           
**                                                            
**                                                            
**                      ###############                       
**                     #:::::::::::::::##                     
**                     #::::::######:::::#                    
**                     #######     #:::::#                    
**     ################            #:::::######   #########   
**   ##:::::::::::::::#            #:::::##::::###:::::::::#  
**  #:::::::::::::::::#         ####::::# #:::::::::::::::::# 
** #:::::::######:::::#    #####::::::##  ##::::::#####::::::#
** #::::::#     #######  ##::::::::###     #:::::#     #:::::#
** #:::::#              #:::::#####        #:::::#     #:::::#
** #:::::#             #:::::#             #:::::#     #:::::#
** #::::::#     ########:::::#             #:::::#    #::::::#
** #:::::::######:::::##:::::#       #######:::::#####:::::::#
**  #:::::::::::::::::##::::::#######:::::##::::::::::::::::# 
**   ##:::::::::::::::##::::::::::::::::::##::::::::::::::##  
**     #####################################::::::########    
**                                         #:::::#            
**                                         #:::::#            
**                                        #:::::::#           
**                                        #:::::::#           
**                                        #:::::::#           
**                                        #########           
**                                                            
** Copyright (c) 1997,2002,2003,2006-2010, azumakuniyuki All rights reserved.
** 
** Redistribution and use in source and binary forms, with or without modification,
** are permitted provided that the following conditions are met:
** 
** Redistributions of source code must retain the above copyright notice, this list
** of conditions and the following disclaimer.
** 
** Redistributions in binary form  must reproduce the above copyright notice,  this
** list of conditions and the following disclaimer in the documentation and/or oth-
** er materials provided with the distribution.
** 
** Neither the name of the ``azumakuniyuki''  nor the names of its contributors may
** be used to endorse or promote products  derived from this software  without spe-
** cific prior written permission.
** 
** THIS SOFTWARE IS PROVIDED  BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
** ANY EXPRESS OR IMPLIED WARRANTIES,  INCLUDING,  BUT NOT LIMITED TO,  THE IMPLIED
** WARRANTIES OF  MERCHANTABILITY  AND FITNESS  FOR A  PARTICULAR PURPOSE  ARE DIS-
** CLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR  CONTRIBUTORS  BE LIABLE FOR
** ANY DIRECT, INDIRECT,  INCIDENTAL, SPECIAL, EXEMPLARY,  OR CONSEQUENTIAL DAMAGES
** ( INCLUDING,  BUT NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE GOODS  OR SERVICES;
** LOSS OF USE, DATA, OR PROFITS;  OR BUSINESS INTERRUPTION ) HOWEVER CAUSED AND ON
** ANY THEORY OF LIABILITY,  WHETHER IN CONTRACT,  STRICT LIABILITY,  OR  TORT (IN-
** CLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFT-
** WARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <getopt.h>
#include <ctype.h>

#include "jp.h"
#include "en.h"

#define SHOW_CONTROLS   ((unsigned int)(1<<0))  /* Show control characters */
#define REQ_WHITESPACE  ((unsigned int)(1<<1))  /* Require whitespace */
#define LANG_JAPANESE   ((unsigned int)(1<<8))  /* Default is Japanese */
#define LANG_ENGLISH    ((unsigned int)(1<<9))  /* Default is English */

#define MAX_LENGTH  127
#define MAX_STDIN   127
#define C2P_OPTIONS "Cd:hL:sv"

/* Default language */
#ifdef JA
#define DEFAULT_LANGUAGE    LANG_JAPANESE
#elif EN
#define DEFAULT_LANGUAGE    LANG_ENGLISH
#else
#define DEFAULT_LANGUAGE    LANG_ENGLISH
#endif

static const char *MY_PROGNAME[] = {"c2p"}; /* My name */
/* static const char *MY_DESCRIPTION[] = {"Convert from character to phonetic code"}; */
static const char *MY_VERSION[] = {"1.1.8"};/* Version number */
static const char CHAR_DELIMITER_C = ',';   /* Delimiter(,) Comma */
static const char CHAR_DELIMITER_P = '|';   /* Delimiter(|) Pipe line */
static const char CHAR_DELIMITER_W = ' ';   /* Delimiter( ) whitespace */
static const char CHAR_DELIMITER_H = '-';   /* Delimiter(-) Hyphen */

extern const char *PHONETIC_JAPANESE[];
extern const char *PHONETIC_ENGLISH[];
static int OPERATION = 0;
static char DELIMITER;

void putHelp(void);
void putVersion(void);

int main( int argc, char *argv[] ) {
    int opt_result;                 /* The result value of the getopt() */
    int num_of_args = argc - 1;     /* The Number Of Arguments(Exclude command itself and options) */
    int statuscode = 0;             /* return(); */
    int givenstring_length = 0;     /* Length of the string */
    short int print_count = 0;      /* Counter for the printing */
    short int check_count = 0;      /* Counter for the checking */
    bool read_from_stdin = 0;       /* If a string came from STDIN, turns on */
    char printablechar_count = 0;   /* Counter for printable characters */
    char multibytechar_count = 0;   /* Counter for multibyte characters */
    char controlchar_count = 0;     /* Counter for contorl characters */

    unsigned char given_characters[MAX_LENGTH] = "";    /* Characters in the argument */
    unsigned char chars_from_stdin[MAX_STDIN] = "";     /* Characters From STDIN */
    unsigned char **selected_language = NULL;           /* Selected language. */
    unsigned char select_char = '\0';
    unsigned char target_char = 0;

    DELIMITER = CHAR_DELIMITER_C;

    while(1) {
        /* Parse options */
        opt_result = getopt( argc, argv, C2P_OPTIONS );
        if( opt_result == -1 ) {
            break;
        }

        switch( opt_result ) {
            case 'C': /* Show control characters */
                OPERATION |= SHOW_CONTROLS;
                num_of_args--;
                break;

            case 'd': /* Select a delimiter */
                switch( *optarg ) {
                    case 'c':   /* Comma */
                        DELIMITER = CHAR_DELIMITER_C;
                        break;
                    case 'h':   /* Hyphen */
                        DELIMITER = CHAR_DELIMITER_H;
                        break;
                    case 'p':   /* Pipe line */
                        DELIMITER = CHAR_DELIMITER_P;
                        break;
                    case 'w':   /* White space */
                        DELIMITER = CHAR_DELIMITER_W;
                        break;
                    default:    /* Noting to do */
                        break;
                }
                num_of_args--;
                break;

            case 'h': /* Help message */
                putHelp();
                exit(EXIT_SUCCESS);
                /* break; */

            case 'L': /* Language */
                switch( *optarg ) {
                    case 'e':   /* English */
                        OPERATION |= LANG_ENGLISH;
                        break;
                    case 'j':   /* Japanese */
                        OPERATION |= LANG_JAPANESE;
                        break;
                    default:
                        break;
                }
                num_of_args--;
                break;

            case 's': /* Delimiter requires a white space after itself. */
                OPERATION |= REQ_WHITESPACE;
                num_of_args--;
                break;

            case 'v': /* Version number */
                putVersion();
                exit(EXIT_SUCCESS);
                /* break; */

            case '?':
            default:
                (void) fprintf( stderr, "try -h option\n" );
                /* break; */
                exit(EXIT_FAILURE);

        } /* end of switch() */

    } /* end of while */


    if( num_of_args < 1 ) {
        /* Read from STDIN, if there is no string in the arguemnts. */
        char __character;
        short int __counter = 0;
        read_from_stdin = 1;

        while( ( __character = getchar() ) != EOF ) {

            if( __counter < ( MAX_STDIN - 1 ) ) {
                chars_from_stdin[__counter] = __character;
                __counter++;

            } else {
                break;
            }
        }

        chars_from_stdin[__counter] = '\0';
        givenstring_length = strlen( (char*) chars_from_stdin );

    } else {
        /* Check the length of the string */
        givenstring_length = strlen( argv[ argc - 1 ] );

        /* Check arguments */
        if( strcmp( argv[ argc - 1 ], argv[0] ) == 0 ) {
            /* There is no argument */
            statuscode = 1;
            return(statuscode);
        }
    }

    if( givenstring_length > MAX_LENGTH ) {
        (void) fprintf( stderr, "Too many characters(%d), max is %d\n", givenstring_length, MAX_LENGTH );
        statuscode = 1;
        return(statuscode);

    } else {
        /* Check selected language */
        if( ( OPERATION & LANG_JAPANESE ) == 0 && ( OPERATION & LANG_ENGLISH ) == 0 ) {
            OPERATION |= DEFAULT_LANGUAGE;
        }

        /* Select a language */
        if( OPERATION & LANG_JAPANESE ) {
            selected_language = (unsigned char **) PHONETIC_JAPANESE;

        } else if( OPERATION & LANG_ENGLISH ) {
            selected_language = (unsigned char **) PHONETIC_ENGLISH;
        }


        /* Copy The String */
        if( read_from_stdin == 1 ) {
#ifdef NOSTRL       /* If 'strlcpy' does not exist */
            strcpy( given_characters, chars_from_stdin );
#else           /* 'strlcpy' exists */
            strlcpy( (char*) given_characters, (char*) chars_from_stdin, sizeof(given_characters) );
#endif
        } else {
#ifdef NOSTRL       /* If 'strlcpy' does not exist */
            strcpy( given_characters, argv[argc-1]);
#else           /* 'strlcpy' exists */
            strlcpy( (char*) given_characters, (char*) argv[ argc - 1 ], sizeof(given_characters) );
#endif
        }


        for( check_count = 0; check_count < givenstring_length; ++check_count ) {
        /* Count the number of multibyte characters and control characters. */
            if( isprint(given_characters[check_count]) ) {
                ++printablechar_count;

            } else if( iscntrl(given_characters[check_count]) ) {
                ++controlchar_count;

            } else if( given_characters[check_count] > 127 ) {
                ++multibytechar_count;
            }
        }

        for( print_count = 0; print_count < givenstring_length; ++print_count ) {
            select_char = given_characters[ print_count ];

            if( OPERATION & SHOW_CONTROLS ) {
                /* No printable characters are left. */
                if( ( print_count + multibytechar_count ) == givenstring_length ) {
                    break;
                }

            } else {
                if( ( print_count + controlchar_count + multibytechar_count ) == givenstring_length ) {
                    break;
                }
            }


            if( isprint(select_char) ) {
                /* Check the ASCII code.  31 < x < 127 */
                if( select_char < 'A' ) {
                    /* Some symbols < 65 */
                    target_char = select_char;
                    printf( "%s", selected_language[ target_char ] );

                } else {
                    /* Check whether or not a character is upper case. */
                    if( isupper(select_char) ) {
                        /* Upper case */
                        target_char = tolower(select_char) - 26;
                        printf( "%s", selected_language[ target_char ] );
                        printf( "%s", selected_language[0] );

                    } else {
                        /* Lower case, some symbols */
                        target_char = select_char - 26; 
                        printf( "%s", selected_language[ target_char ] );
                    }
                }

                if( ! ( OPERATION & SHOW_CONTROLS ) ) {
                    /* Only non-printable characters are left in the string. */
                    if( ( print_count + controlchar_count + multibytechar_count + 1 ) == givenstring_length ) {
                        printf("\n");
                        break;
                    }

                } else {
                    /* Only multibyte characters are left in the string. */
                    if( ( print_count + multibytechar_count + 1 ) == givenstring_length ) {
                        printf("\n");
                        break;
                    }
                }

                if( print_count + 1 < givenstring_length ) {

                    if( ( OPERATION & REQ_WHITESPACE ) && ( DELIMITER != CHAR_DELIMITER_W ) ) {
                        /* A delimiter character and a white space */
                        printf( "%c%c", DELIMITER, ' ' );

                    } else {
                        printf( "%c", DELIMITER );
                    }

                } else {
                    /* This is the last character. */
                    printf("\n");
                    break;
                }

            } else if( iscntrl(select_char) ) {
                /* Control character is included. */
                if( OPERATION & SHOW_CONTROLS ) {

                    if( select_char < ' ' ) {
                        /* Control Character's ASCII code is less than 0x20(Space) */
                        target_char = select_char;

                    } else {
                        /* Control Character is 'Delete'(127) */
                        target_char = select_char - 26;
                    }
                    printf( "%s", selected_language[ target_char ] );

                    /* Only multibyte characters are left in the string. */
                    if( ( print_count + multibytechar_count + 1 ) == givenstring_length ) {
                        printf("\n");
                        break;
                    }

                } else {
                    /* Only non-Printable characters are left in the string. */
                    if( ( print_count + controlchar_count + multibytechar_count + 1 ) == givenstring_length ) {
                        printf("\n");
                        break;
                    }
                }

                if( print_count + 1 < givenstring_length ) {

                    if( OPERATION & SHOW_CONTROLS ) {

                        if( ( OPERATION & REQ_WHITESPACE ) && ( DELIMITER != CHAR_DELIMITER_W ) ) {
                            /* A delimiter character and a white space */
                            printf( "%c%c", DELIMITER, ' ' );

                        } else {
                            printf( "%c", DELIMITER );
                        }
                    }
                    --controlchar_count;
                    continue;

                } else {
                    printf( "\n" );
                    break;
                }

            } else if( select_char > 127 ) {
                /* There is multibyte character. */
                if( print_count + 1 < givenstring_length ) {
                    --multibytechar_count;
                    continue;

                } else {
                    printf("\n");
                    break;
                }
            }
        } /* End of the loop for() */

    }

    return( statuscode );

} /* end of main */

void putHelp(void) {
    printf( "Usage: %s [Options] String\n", *MY_PROGNAME );
    puts( "Options:" );
    puts( "" );
    puts( " Delimiter:" );
    puts( "  -d<x>  : c = Comma [,]" );
    puts( "         : h = Hyphen [-]" );
    puts( "         : p = Pipeline [|]" );
    puts( "         : w = White space [ ]" );
    puts( "");
    puts( " Language:" );
    puts( "  -L<x>  : e = English" );
    puts( "         : j = Japanese(UTF-8)" );
    puts( "");
    puts( "  -s     : Writes a white space after each delimiter." );
    puts( "  -C     : Show control characters too." );
    puts( "  -h     : Shows help message(This screen)." );
    puts( "  -v     : Shows version number." );
}

void putVersion(void) {
    printf( "%s\n", *MY_VERSION );
}



en.h
/* en.h for English */
const char *PHONETIC_ENGLISH[] = {
    "(LARGE)",                  /* is upper case */
    "[start-of-header]",        /* 01   SOA */
    "[start-of-text]",          /* 02   STX */
    "[end-of-text]",            /* 03   ETX */
    "[end-of-transmission]",    /* 04   EOT */
    "[enquiry]",                /* 05   ENQ */
    "[acknowledgement]",        /* 06   ACK */
    "[bell]",                   /* 07   BEL */
    "[backspace]",              /* 08   BS */
    "[horizontal-tab]",         /* 09   HT */
    "[line-feed]",              /* 10   LF */
    "[vertical-tab]",           /* 11   VT */
    "[form-feed]",              /* 12   FF */
    "[carriage-return]",        /* 13   CR */
    "[shift-out]",              /* 14   SO */
    "[shift-in]",               /* 15   SI */
    "[data-link-escape]",       /* 16   DLE */
    "[device-control1/XON]",    /* 17   DC1 */
    "[device-control2]",        /* 18   DC2 */
    "[device-control3/XOFF]",   /* 19   DC3 */
    "[device-control4]",        /* 20   DC4 */
    "[negative-acknowledgement]",/* 21   NAK */
    "[synchronous-idle]",       /* 22   SYN */
    "[end-of-trans.-block]",    /* 23   ETB */
    "[cancel]",                 /* 24   CAN */
    "[end-of-medium]",          /* 25   EM */
    "[substitute]",             /* 26   SUB */
    "[escape]",                 /* 27   ESC */
    "[file-separator]",         /* 28   FS */
    "[group-separator]",        /* 29   GS */
    "[record-separator]",       /* 30   RS */
    "[unit-separator]",         /* 31   US */
    "space",                    /* 32   SP */
    "exclamation-mark", /* 33   ! */
    "double-quotation", /* 34   " */
    "sharp",            /* 35   # */
    "dollar",           /* 36   $ */
    "percent",          /* 37   % */
    "ampersand",        /* 38   & */
    "apostrophe",       /* 39   ' "prime, apostrophe, single quote mark" */
    "left-parenthesis", /* 40   ( */
    "right-parenthesis",/* 41   ) */
    "asterisk",         /* 42   * */
    "plus-sign",        /* 43   + */
    "comma",            /* 44   , */
    "hyphen",           /* 45   - "minus-sign, hyphen" */
    "period",           /* 46   . */
    "slash",            /* 47   / "slash,solidus" */
    "zero",             /* 48   0 */
    "one",              /* 49   1 */
    "two",              /* 50   2 */
    "three",            /* 51   3 */
    "four",             /* 52   4 */
    "five",             /* 53   5 */
    "six",              /* 54   6 */
    "seven",            /* 55   7 */
    "eight",            /* 56   8 */
    "nine",             /* 57   9 */
    "colon",            /* 58   : */
    "semicolon",        /* 59   ; */
    "less-than",        /* 60   < "less than sign" */
    "equal",            /* 61   = */
    "more-than",        /* 62   > "more than sign" */
    "question-mark",    /* 63   ? */
    "at",               /* 64   @ "at, at mark, at sign" */
    "left-bracket",     /* 91   [ */
    "backslash",        /* 92   \ "yen, yen sign" */
    "right-bracket",    /* 93   ] */
    "circumflex",       /* 94   ^ "spacing circumflex accent" */
    "low-line",         /* 95   _ */
    "back-apostrophe",  /* 96   ` "spacing grave accent back apostrophe " */
    "Alfa",             /* 97   a */
    "Blavo",            /* 98   b */
    "Charlie",          /* 99   c */
    "Delta",            /* 100  d */
    "Echo",             /* 101  e */
    "Foxtrot",          /* 102  f */
    "Golf",             /* 103  g */
    "Hotel",            /* 104  h */
    "India",            /* 105  i */
    "Juliet",           /* 106  j */
    "Kilo",             /* 107  k */
    "Lima",             /* 108  l */
    "Mike",             /* 109  m */
    "November",         /* 110  n */
    "Oscar",            /* 111  o */
    "Papa",             /* 112  p */
    "Quebec",           /* 113  q */
    "Romeo",            /* 114  r */
    "Sierra",           /* 115  s */
    "Tango",            /* 116  t */
    "Uniform",          /* 117  u */
    "Victor",           /* 118  v */
    "Whiskey",          /* 119  w */
    "X-Ray",            /* 120  x */
    "Yankee",           /* 121  y */
    "Zulu",             /* 122  z */
    "left-brace",       /* 123  { */
    "vertical-line",    /* 124  | */
    "right-brace",      /* 125  } */
    "tilde",            /* 126  ~ "tilde accent" */
    "[delete]",         /* 127  DEL */
}; /* End of PHONETIC_ENGLISH  */

